omnibus/package-scripts/agent/postinst

```diff
@@ -96,8 +96,119 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
+    datadog-agent version || echo "No datadog-agent binary version (agent 6)"
+    echo "## $INSTALL_DIR"
+    ls -al $INSTALL_DIR || "No agent installed"
+    echo "## $APP_DIR/Contents/Resources"
+    ls -al "$APP_DIR/Contents/Resources" || echo "No app installed"
+
+    # Determine current user if he is using the Graphical installer
+    INSTALL_USER=$(ps aux | grep "CoreServices/Installer" | grep -v grep | awk '{print $1;}')
+
+    # Otherwise, we hope he is using the install script and try to use this user
+    # If it fails, no choice but to use root :'(
+    if [ -z "$INSTALL_USER" ] || [ "$INSTALL_USER" = "root" ]; then
+        SCRIPT_INSTALL="yes"
+        INSTALL_USER=`cat /tmp/datadog-install-user || echo 'root'`
+        rm -v /tmp/datadog-install-user || true
+    fi
+    echo "INSTALL_USER: $INSTALL_USER"
+
+    echo "# Preparing log dir"
+    chown -vR $INSTALL_USER:admin $LOG_DIR
+    chmod -v 755 $LOG_DIR
+
+    echo "# Installing the app"
+    mv -v "$OPT_APP_DIR" /Applications || echo "App already installed"
+
+    # Set the run directory for the agent
+    mkdir -vp "$RUN_DIR"
+    chown -vR $INSTALL_USER:admin "$RUN_DIR"
+    chmod -v 755 "$RUN_DIR"
+
+    echo "# Copying conf"
+    mkdir -vp $CONF_DIR/checks.d
+
+    if [ -e "/tmp/datadog.conf" ] || [ -e "/tmp/datadog.yaml" ]; then
+        mv -vf /tmp/datadog.conf /tmp/datadog.yaml $CONF_DIR
+        cp -vfR /tmp/conf.d/* $CONF_DIR/conf.d
+        cp -vn /tmp/checks.d/* $CONF_DIR/checks.d
+        rm -vrf /tmp/datadog.conf /tmp/conf.d /tmp/checks.d
+    # Or copying default
+    else
+        sed -E 's/^api_key:$/api_key: APIKEY/' $CONF_DIR/datadog.yaml.example > $CONF_DIR/datadog.yaml
+    fi
+
+    echo "# Setting correct rights on conf"
+    chown -v $INSTALL_USER:admin $CONF_DIR/datadog.yaml
+    chown -vR $INSTALL_USER:admin $CONF_DIR/conf.d $CONF_DIR/checks.d
+
+    # `datadog-agent` command line
+    mkdir -vp /usr/local/bin
+    ln -vs $INSTALL_DIR/bin/agent/agent /usr/local/bin/datadog-agent
+
+    # Link for conf files (let's ease the user's life)
+    USER_HOME=`sudo -Hu $INSTALL_USER sh -c 'echo $HOME'`
+    sudo -Hu $INSTALL_USER mkdir -vp "$USER_HOME/.datadog-agent"
+    rm -vf "$USER_HOME/.datadog-agent/conf.d" "$USER_HOME/.datadog-agent/datadog.yaml" "$USER_HOME/.datadog-agent/checks.d"
+    sudo -Hu $INSTALL_USER ln -vs $CONF_DIR/conf.d "$USER_HOME/.datadog-agent/conf.d"
+    sudo -Hu $INSTALL_USER ln -vs $CONF_DIR/datadog.yaml "$USER_HOME/.datadog-agent/datadog.yaml"
+    sudo -Hu $INSTALL_USER ln -vs $CONF_DIR/checks.d "$USER_HOME/.datadog-agent/checks.d"
+
+    # Error if app not properly installed or root
+    if [ "$INSTALL_USER" = "root" ]; then
+        echo 'INSTALL_USER is set to root, Datadog Agent app has been installed'
+        echo 'but is not configured. Running Datadog Agent as root is not advised!'
+        exit 1
+    fi
+
+    echo "# Configuring the agent as a launchd service for the current user (LaunchAgent)"
```

Comment:
that is correct. I think the use cases of the mac Agent are pretty different from other platforms: I'd expect most users to run the mac agent under one user only on their development machine, and I think it's reasonable to target that for our DMG.

That explains why we run the agent under the user that installed it (instead of a `dd-agent`user) and also set the file permissions on that user, why the config directory is available at `~/.datadog-agent`, etc.

If that assumption is not correct for some users, it shouldn't be too complex to turn the agent install into a system-wide install as long as the agent is used entirely through the command line (and the web UI). But making the systray icon work would probably be more complex (since it'd need to have `sudo` rights to start and stop the agent service).

Reply:
I was asking precisely because as you know we encountered this on windows (not exactly, but pretty close) and it ended up being a bit of a mess. Indeed it would probably be way easier to change that on NIX systems.... I don't have a strong preference but just wanted to highlight this.

Reply:
👍  thanks for pointing this out. I think for now we can keep the same approach as agent 5 for the mac agent, it shouldn't be difficult to change this at some point in the future if it makes more sense (unlike the Windows installer, the per-user/per-machine install is not defined in the package itself but just in the files it ships and the permissions it sets on the installed files, which are under our control)